packages/svelte-docgen/src/parser/type/tuple.test.ts

```diff
@@ -162,6 +209,38 @@ describe("Tuple", () => {
 			      "tuple.svelte",
 			    },
 			  },
+			  "Recursive" => {
+			    "alias": "Recursive",
+			    "kind": "union",
+			    "sources": Set {
+			      "tuple.svelte",
+			    },
+			    "types": [
+			      {
+			        "kind": "number",
+			      },
+			      "[<anon:0>]",
```

Comment:
There's no way to make a type reference `"Recursive"`?

Reply:
I'm guessing the circularity would occur in the _encoding_ part?

Reply:
Something quite complex is happening here. Internally, in TypeScript, a Tuple is treated as a kind of “TypeReference” interface. This means that tuples can inherently construct “anonymous” recursive structures. This is an example of such a case. Since the recursive structure is built anonymously, we need to assign some sort of identifier—`<anon:0>` is just something I came up with…

If we need the representation as it appears in the code rather than the optimized internal representation, I think we might need to traverse Declarations instead of Types.

Reply:
Moving tuples into a type map not only solves the recursive issue but also helps consolidate identical tuple types into a single entry. Otherwise, the same anonymous subtypes could appear repeatedly, causing the output to explode. (I observed this when applying `svelte-docgen` to my library.)